azure-resources/src/test/java/io/opentelemetry/contrib/azure/resource/ResourceComponentProviderTest.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.azure.resource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.sdk.autoconfigure.internal.SpiHelper;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.ComponentProvider;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+public class ResourceComponentProviderTest {
+
+  @Test
+  @SuppressWarnings("rawtypes")
+  void providerIsLoaded() {
```

Comment:
I wonder if there's any good way to do an end-to-end style test for these resource detectors like I did here: https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/samplers/src/test/java/internal/RuleBasedRoutingSamplerComponentProviderTest.java#L39

I.e.:
- Have a chunk of yaml that uses the `azure` resource detector like users would
- Call the SDK methods to parse the YAML and create SDK components from it
- Assert that the resource contains expected attributes

This has the benefit of being able to act as a form of documentation for users. 

The assertions may be difficult though because OpenTelemetrySdk doesn't make it easy to access the resource.

Reply:
great idea - I'll create a follow-up PR for this.

Reply:
Created https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2047